phd-tests/framework/src/guest_os/ubuntu24_04.rs

```diff
@@ -0,0 +1,28 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Guest OS adaptations for Ubuntu 24.04 images. These must be prepped with
+//! a cloud-init disk that is configured with the appropriate user and password.
+
+use super::{CommandSequence, CommandSequenceEntry, GuestOs};
+
+pub(super) struct Ubuntu2404;
+
+impl GuestOs for Ubuntu2404 {
+    fn get_login_sequence(&self) -> CommandSequence {
+        CommandSequence(vec![
+            CommandSequenceEntry::wait_for("ubuntu login: "),
+            CommandSequenceEntry::write_str("root"),
+            CommandSequenceEntry::wait_for(self.get_shell_prompt()),
+        ])
+    }
+
+    fn get_shell_prompt(&self) -> &'static str {
+        "root@ubuntu:~#"
+    }
```

Comment:
this was useful for my inconsistently-configured 24.04 test image, but probably not how any other 24.04 test image would be set up. or at least, it's different from 22.04 (which does not expect to be root). so i'll drop this from the PR and make sure we've got something like "how to get or assemble test images for PHD" written up.

Reply:
is 24.04 actually necessary for any particular reason? IIRC on CI we run these tests against an Alpine image, but if we need to start running certain tests against Ubuntu 24.04, presumably we can start doing that as well?

Reply:
not at this point, this is partially an artifact of at one point using `efibootmgr` (which is not in the base Alpine image) for configuration.

i _did_ notice that `guest_can_adjust_boot_order` does not pass with the Alpine CI image, but does with my 24.04 image. i'm not _exactly_ sure why yet, but i don't think it works out in the direction of "need 24.04 in CI".

Reply:
> make sure we've got something like "how to get or assemble test images for PHD" written up

I'd almost forgotten about it, but a ways back I created the [guest-os-image-configs repo](https://github.com/oxidecomputer/guest-os-image-configs) as a place to put stuff like this.